src/serde.rs

```diff
@@ -0,0 +1,181 @@
+        S: Serializer,
+    {
+        let mut bag = serializer.serialize_seq(Some(self.len()))?;
+
+        for (entry, count) in self.set_iter() {
+            bag.serialize_element(&(entry, count))?;
+        }
+
+        bag.end()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use serde_json;
+
+    #[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
+    struct VeryHelpfulStruct {
+        pub(crate) name: String,
+    }
+
+    #[test]
+    fn format_simple_data() {
+        let vikings: HashBag<String> = ["Einar", "Olaf", "Olaf", "Harald", "Harald", "Harald"]
+            .iter()
+            .map(|s| s.to_string())
+            .collect();
+        let einar = "Einar".to_string();
+        let olaf = "Olaf".to_string();
+        let harald = "Harald".to_string();
+        assert_eq!(vikings.get(&einar), Some((&einar, 1)));
+        assert_eq!(vikings.get(&olaf), Some((&olaf, 2)));
+        assert_eq!(vikings.get(&harald), Some((&harald, 3)));
+        println!("Constructed: {:?}", vikings);
+        let jsonified_vikings: String =
+            serde_json::to_string(&vikings).expect("Unable to convert data to json!");
+        println!("JSON: {}", jsonified_vikings);
+        let reconsituted_vikings: HashBag<String> =
+            serde_json::from_str(&jsonified_vikings).expect("Unable to convert json to hashbag!");
+        println!("From json: {:?}", reconsituted_vikings);
+        assert_eq!(vikings, reconsituted_vikings);
+    }
+
+    #[test]
+    fn format_struct_data() {
+        let vikings: HashBag<VeryHelpfulStruct> =
+            ["Einar", "Olaf", "Olaf", "Harald", "Harald", "Harald"]
+                .iter()
+                .map(|n| VeryHelpfulStruct {
+                    name: n.to_string(),
+                })
+                .collect();
+        let einar = VeryHelpfulStruct {
+            name: "Einar".to_string(),
+        };
+        let olaf = VeryHelpfulStruct {
+            name: "Olaf".to_string(),
+        };
+        let harald = VeryHelpfulStruct {
+            name: "Harald".to_string(),
+        };
+        assert_eq!(vikings.get(&einar), Some((&einar, 1)));
+        assert_eq!(vikings.get(&olaf), Some((&olaf, 2)));
+        assert_eq!(vikings.get(&harald), Some((&harald, 3)));
+        println!("Constructed: {:?}", vikings);
+        let jsonified_vikings: String =
+            serde_json::to_string(&vikings).expect("Unable to convert data to json!");
+        println!("JSON: {}", jsonified_vikings);
+        let reconsituted_vikings: HashBag<VeryHelpfulStruct> =
+            serde_json::from_str(&jsonified_vikings).expect("Unable to convert json to hashbag!");
+        println!("From json: {:?}", reconsituted_vikings);
+        assert_eq!(vikings, reconsituted_vikings);
+    }
+
+    #[test]
+    fn repeat_simple_entries() {
+        let jsonified_vikings: String =
+            "[[\"Einar\",1],[\"Olaf\",1],[\"Olaf\",1],[\"Harald\",2],[\"Harald\",1]]".to_string();
```

Comment:
Just a tip: you can use [raw string literals](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals) to avoid all the `\"` in there :)